internal/core/domain/trade_service.go

```diff
@@ -137,15 +145,19 @@ func (t *Trade) Complete(psetBase64 string, txID string) (*CompleteResult, error
 		return &CompleteResult{OK: false}, nil
 	}
 
-	t.Status = CompletedStatus
 	t.SwapComplete.ID = swapCompleteID
 	t.SwapComplete.Message = swapCompleteMsg
 	t.PsetBase64 = psetBase64
-	t.TxID = txID
+	t.TxID = txHash
```

Comment:
this can be just removed since we already set the TxID in trade.Accept()

Reply:
Done.